core/src/main/java/com/linecorp/armeria/common/HttpRequest.java

```diff
@@ -318,6 +321,32 @@ default MediaType contentType() {
         return headers().contentType();
     }
 
+    /**
+     * Returns a list of {@link LanguageRange}s that are specified in {@link HttpHeaderNames#ACCEPT_LANGUAGE}
+     * in the order of client-side preferences. If the client does not send the header, this will contain only a
+     * wild card {@link LanguageRange}.
+     */
+    @Nullable
+    default List<LanguageRange> acceptLanguages() {
+        return headers().acceptLanguages();
+    }
+
+    /**
+     * Matches the {@link Locale}s supported by the server to
+     * the {@link HttpHeaderNames#ACCEPT_LANGUAGE} and returning the best match
+     * according to client preference. It does this via <s>Basic Filter</s>ing each
+     * {@link LanguageRange} and picking the first match. This is the "classic"
+     * algorithm described in
+     * <a href="https://tools.ietf.org/html/rfc2616#section-14.4">RFC2616 Accept-Language (obsoleted)</a>
+     * and also referenced in <a href="https://tools.ietf.org/html/rfc7231#section-5.3.5">RFC7231 Accept-Language</a>.
+     * @param supportedLocales a {@link Collection} of locales supported by the server.
+     * @return The best matching {@link Locale} or {@code null} if no {@link Locale} matches.
+     */
+    @Nullable
+    default Locale selectLocale(Collection<Locale> supportedLocales) {
```

Comment:
- I still think we should accept `Iterable<Locale>` for users. We could convert it into a `Collection` when it's not.
  - Ditto for `RequestHeaders`
- Would you mind adding `selectLocale(Locale...)` as well? I know you don't really like it but all our methods provide a varargs version :sweat_smile:

Reply:
Like I said, I misinterpreted that comment as a question not a very polite order :smile: